### PR TITLE
0.22.0 blogpost

### PR DESCRIPTION
**Pull request by Poolitzer**
Two ToDos, one in line 31. The other one is the open question if we want to add a new section explaining explicitly what litetex did this release for them to be called master meister?

first todo is for @opusforlife2

### CODE REVIEW

File: _posts/2022-02-19-newpipe-0.22.0.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+<figure class="center small">
+    <img src="{{ site.baseurl }}/img/screenshots/comment_pinned.png" class="img-responsive">
+    <figcaption class="text-center">Look at the pin. LOOK AT IT!</figcaption>
+</figure>
+
+<!-- TODO: never heard of "the big ticket change" before, you sure its international?-->
+- The big ticket change for this release will be very obvious to you the moment you double tap to seek on a video. We now have YouTube-like rewind and fast-forward behaviour! Apart from the cool looking visuals, what this means is that you don't have to keep furiously double tapping to execute every single jump in time. Instead, you initiate the action with a double tap, and then every single tap after that executes another jump in time. This 'single tap window' lasts until the animation finishes, after which you have to double tap to initiate another one. This huge quality-of-life improvement to seeking was brought to you by [@vkay94](https://github.com/vkay94). The PR was kept up to date by [@Stypox](https://github.com/Stypox), and then finally pushed across the finish line by [@litetex](https://github.com/litetex). Thanks to everyone involved in this major effort!
```
Comment (TobiGr) [inline _posts/2022-02-19-newpipe-0.22.0.md:32]: A screenshot would be nice :)

File: _posts/2022-02-19-newpipe-0.22.0.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+- We got another improvement for people with slow Internet or those facing a period of poor connection. The player used to "close with an error" (speak: crash) if the player took too long to buffer during a livestream. This happened because it just fell too far behind the edge of the livestream. To fix and improve this, [@TiA4f8R](https://github.com/TiA4f8R) implemented a two-fold change. The first was to increase the time period before the player is considered to have fallen too far behind, and the second was to simply drop the current play time and jump ahead to the edge of the livestream, skipping the un-buffered section entirely. No more player crashes! Never! Right guys?
+
+### Fixed
+
+- As the first entry in our big series of fixes this release, [@litetex](https://github.com/litetex) disabled an unreliable test for the SoundCloud stream extractor. Who even needs tests? (Just kidding, they're a cornerstone of good programming. ~~Eat your broccoli~~ Write your unit tests, kids.)
```
Comment (TobiGr) [inline _posts/2022-02-19-newpipe-0.22.0.md:49]: This belongs to the Development / nerd talk section

File: _posts/2022-02-19-newpipe-0.22.0.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+  - Also, the `Clear reCAPTCHA cookies` option was moved from `Content` to `History and cache`, and changed to being greyed out by default instead of being hidden by default, so that it can be shown in the search results.
+
+### Improved
+
+- Are you tired of the toast notifications telling you that an error occurred, but having no idea what happened, or how to report that error? Well, in addition to the error report page and the error snackbar at the bottom, now NewPipe has a 3rd way to show errors: it generates a silent system notification. You just need to tap on it to report the error to us, all thanks to [@Stypox](https://github.com/Stypox), who also took the opportunity to restructure and improve the rest of the error reporting related code.
```
Comment (mhmdanas) [inline _posts/2022-02-19-newpipe-0.22.0.md:39]:
```suggestion
- Are you tired of the toast notifications telling you that an error occurred, but having no idea what happened, or how to report that error? Well, in addition to the error report page and the error snackbar at the bottom, now NewPipe has a third way to show errors: it generates a silent system notification. You just need to tap on it to report the error to us, all thanks to [@Stypox](https://github.com/Stypox), who also took the opportunity to restructure and improve the rest of the error reporting related code.
```

File: _posts/2022-02-19-newpipe-0.22.0.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+### Fixed
+
+- As the first entry in our big series of fixes this release, [@litetex](https://github.com/litetex) disabled an unreliable test for the SoundCloud stream extractor. Who even needs tests? (Just kidding, they're a cornerstone of good programming. ~~Eat your broccoli~~ Write your unit tests, kids.)
+
+- When a video link was opened with a timestamp URL parameter that had more than 3 digits, NewPipe ignored the 4th and higher digits. This meant that the app took you to the wrong timestamp. For example, 1001 seconds (16 minutes and 41 seconds) would be parsed as 1 second instead. [@mhmdanas](https://github.com/mhmdanas) has fixed this.
+
+- A nasty bug a lot of our users faced was that switching from one player type to another caused the playback to restart from the beginning. This has been fixed by [@litetex](https://github.com/litetex).
```
Comment (TobiGr) [inline _posts/2022-02-19-newpipe-0.22.0.md:53]: That's a high priority bug, I'd put it at the beginning of the section

File: _posts/2022-02-19-newpipe-0.22.0.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+- [@Stypox](https://github.com/Stypox) fixed the order of local search results from your search history, so now they are back to being ordered by creation date instead of alphabetical order (which made it seem like new entries weren't being added to the search result history database, unless you searched for a video titled "AAAA" or something).
+
+ -[@XiangRongLin](https://github.com/XiangRongLin) ensured that search history entries in tests are in the correct order, fixing the sometimes-failing-sometimes-passing CI pipeline.
```
Comment (TobiGr) [inline _posts/2022-02-19-newpipe-0.22.0.md:57]: This is also development stuff.
```suggestion
- [@XiangRongLin](https://github.com/XiangRongLin) ensured that search history entries in tests are in the correct order, fixing the sometimes-failing-sometimes-passing CI pipeline.
```

File: _posts/2022-02-19-newpipe-0.22.0.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+- [@Stypox](https://github.com/Stypox) fixed many different crashes where the code expected the player's queue to be not `null` but it was, indeed, `null`.
+
+- [@Cheechaii](https://github.com/Cheechaii) introduced #respect into the app. Now the player respects display cutouts (like notches) when playing in multi-window mode, whereas previously, black bars were shown.
+
+- Speaking of hashtags, [@Jaspann](https://github.com/Jaspann) made their first contribution to a public open source project (Congratulations!): They fixed hashtags that use non-English characters. Thank you!
+
+- [@Jaspann](https://github.com/Jaspann), in their second contribution, fixed showing the PeerTube-specific sub-channel icon on non-PeerTube services, where it makes no sense.
+
+- [@TobiGr](https://github.com/TobiGr) fixed a link in the English README linking to the Portuguese README. Not really their first/second contribution to the project, a welcome one nonetheless
```
Comment (TobiGr) [inline _posts/2022-02-19-newpipe-0.22.0.md:67]: Really unimportant, I'd move it down as much as possible, maybe into the localisation section

File: _includes/release_data.html [hunk 7]
```diff
@@ -10,10 +10,13 @@
 }}
 {%- capture releaseChangelog -%}
 <div>
-    <p>Fixed YouTube not playing any stream.</p>
+    <p>YouTube-like rewind/fast-forward</p>
+    <p>Pinned comments</p>
+    <p>Searchable preferences</p>
+    <p>Fixed player playback reset, order of search history, opening video links with 4 digits+ timestamps, and more</p>
```
Comment (TobiGr) [inline _includes/release_data.html:16]: The whole `releaseChangelog` var should be:
``` HTML
<ul>
    <li>YouTube-like rewind/fast-forward</li>
    <li>Pinned comments</li>
    <li>Searchable preferences</li>
    <li>Fixed player playback reset, order of search history, opening video links with 4 digits+ timestamps, and more</li>
</ul>
```

**Review by TobiGr — COMMENTED** [review]
Thank you :heart:

File: _posts/2022-02-19-newpipe-0.22.0.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+  - PrettyTime 5.0.1.Final -> 5.0.2.Final
+  - AndroidX Test JUnit 1.1.2 -> 1.1.3
+
+- [@XiangRongLin](https://github.com/XiangRongLin) added the SonarList CI job back to the test run setup, which had been disabled earlier due to some issues.
+
+- [@TobiGr](https://github.com/TobiGr) made some small improvements to code quality and readability here and there.
+
+- [@litetex](https://github.com/litetex) configured the CI to upload test reports when an error occurs, because the truncated stacktrace shown by default doesn't provide enough info sometimes.
+
+- [@XiangRongLin](https://github.com/XiangRongLin) switched the JUnit asserts with AssertJ ones in HistoryRecordManagerTest to obtain more detailed error messages, and converted the `SearchHistoryEntry` Java class to a Kotlin data class for the same reason.
```
Comment (mhmdanas) [inline _posts/2022-02-19-newpipe-0.22.0.md:124]:
```suggestion
- [@XiangRongLin](https://github.com/XiangRongLin) switched the JUnit asserts with AssertJ ones in `HistoryRecordManagerTest` to obtain more detailed error messages, and converted the `SearchHistoryEntry` Java class to a Kotlin data class for the same reason.
```

File: _posts/2022-02-19-newpipe-0.22.0.md [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+
+- [@mhmdanas](https://github.com/mhmdanas) fixed a false Android Studio warning by replacing a direct.
```
Comment (mhmdanas) [inline _posts/2022-02-19-newpipe-0.22.0.md:126]: This isn't complete.

**Review by mhmdanas — CHANGES_REQUESTED** [review]

File: _posts/2022-02-19-newpipe-0.22.0.md (hunk 4 above)
Comment (mhmdanas) [reply _posts/2022-02-19-newpipe-0.22.0.md:53]: I think the items are in order of first merged to most recently merged, which I think isn't exactly good in release blog posts?

File: _posts/2022-02-19-newpipe-0.22.0.md [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+
+- If you tried to share a playlist while it was still loading (you maniac) it would cause the app to crash, which has been fixed by [@TiA4f8R](https://github.com/TiA4f8R).
+
+- [@Stypox](https://github.com/Stypox) added the app icon to the fastlane metadata folder, so that the F-Droid app and website start showing the icon again.
+
+- Ooh. This is a tricky one. If a stream other than the first one in the queue was playing, and its resolution was changed, or the player type was changed (e.g. popup -> main player), the app would suddenly abandon that stream and start playing the first stream in the queue instead. This weird bug, caused by an ExoPlayer update, has been fixed by [@TiA4f8R](https://github.com/TiA4f8R).
+
+- If you tried to view the channel page of an item in the queue whose channel page you hadn't already viewed before, you would get a network error, which has been fixed by [@ktprograms](https://github.com/ktprograms).
+
+## Localisation
```
Comment (mhmdanas) [inline _posts/2022-02-19-newpipe-0.22.0.md:81]: Where is https://github.com/TeamNewPipe/NewPipe/pull/7694?

Comment (Poolitzer) [reply _posts/2022-02-19-newpipe-0.22.0.md:81]: Quite sure that wasn't in the draft, so I want to point the blame to someone else, but yes. Needs to be added ofc.

Comment (mhmdanas) [reply _posts/2022-02-19-newpipe-0.22.0.md:81]: > so I want to point the blame to someone else

That would be me. xD

File: _posts/2022-02-19-newpipe-0.22.0.md (hunk 3 above)
Comment (Poolitzer) [reply _posts/2022-02-19-newpipe-0.22.0.md:39]: agreed better!

File: _posts/2022-02-19-newpipe-0.22.0.md (hunk 4 above)
Comment (Poolitzer) [reply _posts/2022-02-19-newpipe-0.22.0.md:53]: no its a "how one feels" order, I already put it up a lot. Will put it up front.

File: _posts/2022-02-19-newpipe-0.22.0.md (hunk 5 above)
Comment (Poolitzer) [reply _posts/2022-02-19-newpipe-0.22.0.md:57]: The idea behind this was to group them together (thats why its a subpoint), I will move it down.

File: _posts/2022-02-19-newpipe-0.22.0.md (hunk 6 above)
Comment (Poolitzer) [reply _posts/2022-02-19-newpipe-0.22.0.md:67]: So now you destroyed that joke opportunity pfff.

File: _posts/2022-02-19-newpipe-0.22.0.md (hunk 9 above)
Comment (Poolitzer) [reply _posts/2022-02-19-newpipe-0.22.0.md:126]: It isn't, no idea how that escaped me.

File: _posts/2022-02-19-newpipe-0.22.0.md [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+Apart from these, we have a bucket-load of fixes and improvements that will hopefully make your NewPipe experience better than ever.
+
+<!-- more -->
+
+This blog post is dedicated to the master meister [@litetex](https://github.com/litetex). Read on and it will be quite evident why.
+
+## New
+
+- We added support for opening another PeerTube instance fsuggestions from code reviewrom outside the app: [FSFE-Tube](https://media.fsfe.org/). Enjoy the videos there, and thank [@mhmdanas](https://github.com/mhmdanas) for implementing this.
+suggestions from code review
```
Comment (mhmdanas) [inline _posts/2022-02-19-newpipe-0.22.0.md:23]: What is this?

**Review by mhmdanas — CHANGES_REQUESTED** [review]

**Review by TobiGr — APPROVED** [review]

File: _posts/2022-02-19-newpipe-0.22.0.md [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+
+- dependabot updates:
+  
+  - Bump spotbugs-annotations from 4.5.0 to 4.5.3
+
+- [@litetex](https://github.com/litetex) disabled an unreliable test for the SoundCloud stream extractor. Who even needs tests? (Just kidding, they're a cornerstone of good programming. ~~Eat your broccoli~~ Write your unit tests, kids.)
+
+- [@XiangRongLin](https://github.com/XiangRongLin) ensured that search history entries in tests are in the correct order, fixing the sometimes-failing-sometimes-passing CI pipeline. Remember what we said about tests above, they are important!
+
+- YouTube removed the public dislike data [in December](https://support.google.com/youtube/thread/134791097/update-to-youtube-dislike-counts). So we used a temporary workaround by calculating the dislikes using the average rating. Now that is gone too. So [@litetex](https://github.com/litetex) removed the code used to extract YouTube dislikes, and made several other improvements:
```
Comment (mhmdanas) [inline _posts/2022-02-19-newpipe-0.22.0.md:102]:
```suggestion
- YouTube removed the public dislike data [in December](https://support.google.com/youtube/thread/134791097/update-to-youtube-dislike-counts), so we used a temporary workaround by calculating the dislikes using the average rating. Now that is gone too, so [@litetex](https://github.com/litetex) removed the code used to extract YouTube dislikes, and made several other improvements:
```

**Review by mhmdanas — CHANGES_REQUESTED** [review]

File: _posts/2022-02-19-newpipe-0.22.0.md [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+
+- Talking about broken links, [@TobiGr](https://github.com/TobiGr) fixed a link in the English README linking to the Portuguese README.
+
+- [@albertomosconi](https://github.com/albertomosconi) moved the translated README files into the `doc` folder, and updated the links accordingly.
+
+## Nerd Talk
+
+- dependabot updates:
+  
+  - Bump spotbugs-annotations from 4.5.0 to 4.5.3
```
[the remark this answers is not in the document]
Comment (TobiGr) [reply _posts/2022-02-19-newpipe-0.22.0.md:96]: What about 
```suggestion
- Dependabot updated spotbugs-annotations from 4.5.0 to 4.5.3.
```

File: _posts/2022-02-19-newpipe-0.22.0.md [hunk 14]
```diff
@@ -0,0 +1,162 @@
...
+
+- Ooh. This is a tricky one. If a stream other than the first one in the queue was playing, and its resolution was changed, or the player type was changed (e.g. popup -> main player), the app would suddenly abandon that stream and start playing the first stream in the queue instead. This weird bug, caused by an ExoPlayer update, has been fixed by [@TiA4f8R](https://github.com/TiA4f8R).
+
+- If you tried to view the channel page of an item in the queue whose channel page you hadn't already viewed before, you would get a network error, which has been fixed by [@ktprograms](https://github.com/ktprograms).
+
+## Localisation
+
+- Polish was added as a new translation by [@Radplay](https://github.com/Radplay).
+
+- You want more languages? Well we also have Traditional Chinese (used in Taiwan) now, added by [@akamayu-ouo](https://github.com/akamayu-ouo). Thank you!
```
[the remark this answers is not in the document]
Comment (TobiGr) [reply _posts/2022-02-19-newpipe-0.22.0.md:83]:
```suggestion
- A Polish translation of the README was added by [@Radplay](https://github.com/Radplay). You want more languages? Well we also have one in Traditional Chinese (used in Taiwan) now, added by [@akamayu-ouo](https://github.com/akamayu-ouo). Thank you!
```

File: _posts/2022-02-19-newpipe-0.22.0.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (mhmdanas) [reply _posts/2022-02-19-newpipe-0.22.0.md:96]: Ah, now I got what it means, I didn't understand initially, either is fine. If you can't choose, I'd go for your suggestion.

File: _posts/2022-02-19-newpipe-0.22.0.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (mhmdanas) [reply _posts/2022-02-19-newpipe-0.22.0.md]: Looks good, just split it into two items:
```suggestion
- A Polish translation of the README was added by [@Radplay](https://github.com/Radplay). Thank you!

- You want more languages? Well, we also have one in Traditional Chinese (used in Taiwan) now, added by [@akamayu-ouo](https://github.com/akamayu-ouo). Thank you too!
```

**Review by mhmdanas — APPROVED** [review]

**Comment by opusforlife2** [comment]
> I think the items are in order of first merged to most recently merged, which I think isn't exactly good in release blog posts?

@mhmdanas Replying here because GH notifs don't take you to review comments: The items are ordered in that way simply because it's convenient to compare the blog post with the release notes. Once the blog post is done, there is no issue with reordering them.

**Pull request merged**